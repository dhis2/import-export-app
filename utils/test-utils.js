import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
