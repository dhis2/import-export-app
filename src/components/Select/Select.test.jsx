import React from 'react'
import { render } from 'test-utils'

import { Select } from '../index.js'

const options = [
    { value: 'a', label: 'Abc' },
    { value: 'b', label: 'Def' },
    { value: 'c', label: 'Ghi' },
]

const setValue = jest.fn().mockName('setValue')

const props = {
    name: 'object-type',
    label: 'Object type',
    options: options,
    checked: options[0],
    setValue: setValue,
    meta: {},
    dataTest: 'select-object-type',
}

it('matches snapshot', () => {
    const { asFragment } = render(<Select {...props} />)
    expect(asFragment()).toMatchSnapshot()
})
