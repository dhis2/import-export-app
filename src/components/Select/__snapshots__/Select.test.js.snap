// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    data-test="select-object-type"
  >
    <span
      class="label"
    >
      Object type
    </span>
    <div
      class="select"
      data-test="select-object-type-container"
    >
      <div
        class="jsx-1141087211 "
        data-test="select-object-type-ssf"
      >
        <div
          class="jsx-3375842594 "
          data-test="select-object-type-ssf-content"
        >
          <div
            class="jsx-3415779570 "
            data-test="dhis2-uicore-box"
          >
            <div
              class="jsx-114080822 root"
              data-test="dhis2-uicore-singleselect"
            >
              <div
                class="jsx-114080822 root-input"
              >
                <div
                  data-test="dhis2-uicore-select"
                >
                  <div
                    class="jsx-1165050001 root"
                    data-test="dhis2-uicore-select-input"
                    tabindex="0"
                  >
                    <div
                      class="jsx-1165050001 root-children"
                    >
                      <div
                        class="jsx-2240447017 jsx-4123978364 root"
                      />
                    </div>
                    <div
                      class="jsx-1165050001 root-right"
                    >
                      <svg
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.146 6.146a.5.5 0 01.638-.057l.07.057L8 9.293l3.146-3.147a.5.5 0 01.638-.057l.07.057a.5.5 0 01.057.638l-.057.07-3.5 3.5a.5.5 0 01-.638.057l-.07-.057-3.5-3.5a.5.5 0 010-.708z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
